src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] hashTable;
+    private int threshold = DEFAULT_INITIAL_CAPACITY * (int) DEFAULT_LOAD_FACTOR;
+    private int size;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold || hashTable == null) {
+            resize();
+        }
+        putValue(indexFromHash(key), key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (hashTable != null) {
+            int index = indexFromHash(key);
+            Node<K, V> currentNode = hashTable[index];
+            while (currentNode != null) {
+                if (Objects.equals(key, currentNode.getKey())) {
+                    return currentNode.getValue();
+                }
+                currentNode = currentNode.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
```

Comment:
private classes and methods should be at the bottom of the class.
Also, nested classes do not need getters and setters, you can reference their fields directly